### PR TITLE
LPS-90943 Add data record collection search feature

### PR DESCRIPTION
**Pull request by mclomello**
@gabrielwas

### CODE REVIEW

File: modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/service/DEDataRecordCollectionServiceImpl.java [hunk 1]
```diff
@@ -577,6 +598,27 @@ protected DEDataEnginePermissionSupport getDEDataEnginePermissionSupport() {
 		return new DEDataEnginePermissionSupport(groupLocalService);
 	}
 
+	protected DEDataEngineRequestExecutor getDEDataEngineRequestExecutor() {
```
Comment (gabrielwas) [inline modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/service/DEDataRecordCollectionServiceImpl.java]: @mclomello Please, remove this hunk (methods getDEDataEngineRequestExecutor and getDeDataRecordCollectionSearchExecutor.

File: modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/service/DEDataRecordCollectionServiceImpl.java [hunk 2]
```diff
@@ -480,6 +483,24 @@ public DEDataRecordCollectionSaveResponse execute(
 		}
 	}
 
+	@Override
```
Comment (gabrielwas) [inline modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/service/DEDataRecordCollectionServiceImpl.java]: @mclomello Remove deDataRecordCollectionSearchExecutor instance.
```suggestion
	@Override
	public DEDataRecordCollectionSearchResponse execute(
			DEDataRecordCollectionSearchRequest deDataRecordCollectionSearchRequest)
		throws DEDataRecordCollectionException {

		try {
			return deDataRecordCollectionSearchExecutor.execute(
				deDataRecordCollectionSearchRequest);
		}
		catch (Exception e) {
			throw new DEDataRecordCollectionException(e);
		}
	}
```

File: modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/service/DEDataRecordCollectionServiceImpl.java [hunk 3]
```diff
@@ -480,6 +483,24 @@ public DEDataRecordCollectionSaveResponse execute(
...
+	public DEDataRecordCollectionSearchResponse execute(
+			DEDataRecordCollectionSearchRequest deDataRecordCollectionSearchRequest)
+		throws DEDataRecordCollectionException {
+
+		DEDataRecordCollectionSearchExecutor
+			deDataRecordCollectionSearchExecutor =
+				getDeDataRecordCollectionSearchExecutor();
+
+		try {
+			return deDataRecordCollectionSearchExecutor.execute(
```
Comment (gabrielwas) [inline modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/service/DEDataRecordCollectionServiceImpl.java]: @mclomello Change the variable
Sugestion:

```suggestion
			return _deDataRecordCollectionSearchExecutor.execute(
```

File: modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/service/DEDataRecordCollectionServiceImpl.java (hunk 1 above)
Comment (mclomello) [reply modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/service/DEDataRecordCollectionServiceImpl.java]: Yes Gabriel. Like we talk earlier, I remove and adjust my implementation to work with the new approach.

File: modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/service/DEDataRecordCollectionServiceImpl.java (hunk 2 above)
Comment (mclomello) [reply modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/service/DEDataRecordCollectionServiceImpl.java]: Changed and fixed according last comment.

File: modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/service/DEDataRecordCollectionServiceImpl.java (hunk 3 above)
Comment (mclomello) [reply modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/service/DEDataRecordCollectionServiceImpl.java]: Fixed according by previous comments.

**Comment by mclomello** [comment]
ci:test:sf

**Comment by gabrielwas** [comment]
ci:test:sf

**Comment by gabrielwas** [comment]
ci:test:relevant

**Comment by mclomello** [comment]
ci:test:sf

**Comment by gabrielwas** [comment]
ci:test:sf

**Comment by gabrielwas** [comment]
ci:test:relevant

**Comment by gabrielwas** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by gabrielwas** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/68769. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**